src/tyr/route_serializer_valhalla.cc

```diff
@@ -538,15 +537,15 @@ void legs(const valhalla::Api& api, int route_index, rapidjson::writer_wrapper_t
 std::string serialize(const Api& api) {
   // build up the json object, reserve 4k bytes
   rapidjson::writer_wrapper_t writer(4096);
-  writer.start_object();
 
   // for each route
   for (int i = 0; i < api.directions().routes_size(); ++i) {
     if (i == 1) {
       writer.start_array("alternates");
     }
 
-    // the main route
+    // the route itself
+    writer.start_object();
```

Comment:
every route needs to be wrapped in an object `{` we had only one for the outer object in the response but for each of the alternates we didnt add one. this lead to json that looked like:

```json
{
  "trip": .... main route here,
  "alternates": [
    "trip",
    {guts of alternate here},
```